lib/library.rb

```diff
@@ -0,0 +1,50 @@
+class Library
+  attr_reader :name, :books, :authors
+  def initialize(name)
+    @name = name
+    @books_array = []
+    @authors = []
+  end
+
+  def add_author(author)
+    @authors.push author
+  end
+
+  def books
+    @authors.each do |author|
+        author.books.each do |book|
+          book_being_added = book
+          @books_array.push book_being_added
```

Comment:
Here, you could push `book` rather than assigning book to another variable; but, the logic here is super solid.  I would do something similar 🙂